Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -115,6 +115,25 @@ TEST_CASE("URL Generation") {
 }
 
 
+TEST_CASE_METHOD(ReplicatorAPITest, "API Create C4Replicator without start", "[Push]") {
+    // For CBL-524 "Lazy c4replicator initialize cause memory leak"
+    createDB2();
+
+    C4Error err;
+    C4ReplicatorParameters params = {};
+    params.push = kC4OneShot;
+    params.pull = kC4Disabled;
+    params.callbackContext = this;
+    params.socketFactory = _socketFactory;
+    params.dontStart = true;
+
+    _repl = c4repl_new(db, _address, kC4SliceNull, (C4Database*)db2,
+                       params, &err);
+    C4Log("---- Releasing C4Replicator ----");
+    _repl = nullptr;
```

Comment:
Not sure this will add to the test since this will get released later.  Is there something visible that happens when it is done early here?

Reply:
Mostly I just wanted to have an explicit, visible place where it's being released, rather than down inside auto-generated code in the test destructor. It made it easier to set breakpoints.